dali/pipeline/operators/geometric/bb_flip_test.cc

```diff
@@ -0,0 +1,230 @@
+                std::tuple<Roi, Roi, Roi>{{.25, .25, .5, .5},
+                                          {.25, .25, .5, .5},
+                                          {.25, .25, .5, .5}},
+        }
+};
+
+constexpr TestData two_pt_rois{
+        {
+                std::tuple<Roi, Roi, Roi>{{.2, .2, .6, .5},
+                                          {.4, .2, .8, .5},
+                                          {.2, .5, .6, .8}},
+                std::tuple<Roi, Roi, Roi>{{.0, .0, .5, .5},
+                                          {.5, .0, 1., .5},
+                                          {.0, .5, .5, 1.}},
+                std::tuple<Roi, Roi, Roi>{{.3, .2, .4, .3},
+                                          {.6, .2, .7, .3},
+                                          {.3, .7, .4, .8}},
+                std::tuple<Roi, Roi, Roi>{{.0, .0, .2, .3},
+                                          {.8, .0, 1., .3},
+                                          {.0, .7, .2, 1.}},
+                std::tuple<Roi, Roi, Roi>{{.0, .0, .1, .1},
+                                          {.9, .0, 1., .1},
+                                          {.0, .9, .1, 1.}},
+                std::tuple<Roi, Roi, Roi>{{.5, .5, .6, .6},
+                                          {.4, .5, .5, .6},
+                                          {.5, .4, .6, .5}},
+                std::tuple<Roi, Roi, Roi>{{.0, .6, .7, .9},
+                                          {.3, .6, 1., .9},
+                                          {.0, .0, .7, .4}},
+                std::tuple<Roi, Roi, Roi>{{.6, .2, .3, .3},
+                                          {.1, .2, .4, .5},
+                                          {.6, .5, .9, .8}},
+                std::tuple<Roi, Roi, Roi>{{.4, .3, .9, .8},
+                                          {.1, .3, .6, .8},
+                                          {.4, .2, .9, .7}},
+                std::tuple<Roi, Roi, Roi>{{.25, .25, .75, .75},
+                                          {.25, .25, .75, .75},
+                                          {.25, .25, .75, .75}},
+        }
+};
+
+
+/**
+ * Injects either input values of test std::arrays (i.e. left-hand
+ * Rois) or anticipated output values, which is the reference data.
+ * @param input If true, left-hand Rois will be injected.
+ */
+template<typename DataType, int DataIdx>
+void InjectTestData(const TestData &test_data, DataType *destination) {
+  for (const auto &it : test_data) {
+    auto _it = std::get<DataIdx>(it).data();
+    std::memcpy(destination, _it, kBbStructSize * sizeof(DataType));
+    destination += kBbStructSize;
+  }
+}
+
+}  // namespace
+
+template<typename ImageType>
+class BbFlipTest : public DALISingleOpTest<ImageType> {
+ protected:
+  std::vector<TensorList<CPUBackend> *>
+  Reference(const std::vector<TensorList<CPUBackend> *> &inputs, DeviceWorkspace *ws) override {
+    using DataType = float;
+    auto batch = new TensorList<CPUBackend>();
```

Comment:
auto *batch = ..., just like below?

Reply:
I'm not sure I understand what you mean here. Could you please amplify?